src/data/matrix/basic.lean

```diff
@@ -405,9 +439,15 @@ variables [non_unital_non_assoc_semiring α]
 @[simp] protected theorem mul_zero (M : matrix m n α) : M ⬝ (0 : matrix n o α) = 0 :=
 by { ext i j, apply dot_product_zero }
 
+@[simp] protected theorem mul_zero' (M : matrix m n α) : M ⬝ (λ i j , 0 : matrix n o α) = 0 :=
+by { ext i j, apply dot_product_zero }
+
 @[simp] protected theorem zero_mul (M : matrix m n α) : (0 : matrix l m α) ⬝ M = 0 :=
 by { ext i j, apply zero_dot_product }
 
+@[simp] protected theorem zero_mul' (M : matrix m n α) : (λ i j , 0 : matrix l m α) ⬝ M = 0 :=
+by { ext i j, apply zero_dot_product }
+
```

Comment:
I don't think we want these lemmas. If you have `(λ i j, 0 : matrix l m α)` in your goals state then something has likely gone wrong. Are there other examples like this in the file that you're copying?

Reply:
Yes, there are. I wrote these two as I saw
```
@[simp] lemma dot_product_zero' : dot_product v (λ _, 0) = 0 := dot_product_zero v
@[simp] lemma zero_dot_product' : dot_product (λ _, (0 : α)) v = 0 := zero_dot_product v
```

Reply:
Those are different, because `λ _, 0` is not a matrix but a vector. We have a dedicated type for matrices, so `(λ i j, 0 : matrix l m α)` should never appear. We do not have a dedicated type for vectors. so those lemmas are ok.

Reply:
OK. I see the difference. If they are not useful, I will delete them.